### PR TITLE
[#140641805] Accordion

### PR DESCRIPTION
**Pull request by KernCheh**
Hey @zamakkat 

There you go, accordion

https://ebon-chalice.luxola.com/style-guide

### CODE REVIEW

File: app/assets/javascripts/sephora_style_guide/application.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 //= require 'jquery'
 //= require_self
+//= require './transition'
```
Comment (zamakkat) [inline app/assets/javascripts/sephora_style_guide/application.js]: There is already a require_tree right below so you shouldn't need this. Also that whole file is commented out?

File: app/assets/stylesheets/sephora_style_guide/mixins/_border_radius.scss [hunk 2]
```diff
@@ -0,0 +1,33 @@
+@mixin border-radius($radius: $border-radius) {
```
Comment (zamakkat) [inline app/assets/stylesheets/sephora_style_guide/mixins/_border_radius.scss]: AFIK we rarely have designs with rounded borders so I didn't include these initially.

File: app/assets/stylesheets/sephora_style_guide/components/_card.scss [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+// Remove all backgrounds
+.card-outline-primary {
+  @include card-outline-variant($btn-primary-bg);
+}
+
+//
+// Inverse text within a card for use with dark backgrounds
+//
+
+.card-inverse {
```
Comment (zamakkat) [inline app/assets/stylesheets/sephora_style_guide/components/_card.scss]: Just make sure stuff is added selectively. If something suggests we will never use it then it's fine to remove from the default Bootstrap.

File: app/assets/stylesheets/sephora_style_guide/variables/_border.scss [hunk 4]
```diff
@@ -0,0 +1 @@
+$border-radius:          .25rem !default;
```
Comment (zamakkat) [inline app/assets/stylesheets/sephora_style_guide/variables/_border.scss]: Prefer pixels

File: app/assets/stylesheets/sephora_style_guide/variables/_grid.scss [hunk 5]
```diff
@@ -19,3 +19,4 @@ $grid-gutter-widths: (
   tn: 20px,
   sm: 30px,
 );
+
```
Comment (zamakkat) [inline app/assets/stylesheets/sephora_style_guide/variables/_grid.scss]: Extra line

File: app/assets/stylesheets/sephora_style_guide/variables/_typography.scss [hunk 6]
```diff
@@ -12,3 +12,5 @@ $line-height-root: 16px;
 $line-height-root-mobile: 20px;
 
 $line-height-base: 16px;
+
+$line-height-computed:    floor(($font-size-base * 1.428571429)) !default;
```
Comment (zamakkat) [inline app/assets/stylesheets/sephora_style_guide/variables/_typography.scss]: Not being used?

File: app/views/sephora_style_guide/home/_accordion.slim [hunk 7]
```diff
@@ -0,0 +1,32 @@
+article
+
+  h2.h2 Accordion
```
Comment (zamakkat) [inline app/views/sephora_style_guide/home/_accordion.slim]: Would just prefer a more consistent naming. Either call it a collapse or accordion, as long as it's used in all CSS classes and namings.

File: app/views/sephora_style_guide/home/_accordion.slim [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+      #headingOne.card-header.inverse role="tab"
+        h3.h3.mb-0
+          a aria-controls="collapseOne" aria-expanded="true" data-parent="#accordion" data-toggle="collapse" href="#collapseOne"
+            | Collapsible Group Item #1
+      #collapseOne.collapse aria-labelledby="headingOne" role="tabpanel"
+        .card-block
+          | Mr. and Mrs. Dursley, of number four, Privet Drive, were proud to say that they were perfectly normal, thank you very much. They were the last people you’d expect to be involved in anything strange or mysterious, because they just didn’t hold with such nonsense.
+
+    .card
+      #headingTwo.card-header role="tab"
```
Comment (zamakkat) [inline app/views/sephora_style_guide/home/_accordion.slim]: Prefer single quotes

File: app/views/sephora_style_guide/home/_accordion.slim [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+        h3.h3.mb-0
+          a.collapsed aria-controls="collapseTwo" aria-expanded="false" data-parent="#accordion" data-toggle="collapse" href="#collapseTwo"
+            | Collapsible Group Item #2
+      #collapseTwo.collapse aria-labelledby="headingTwo" role="tabpanel"
```
Comment (zamakkat) [inline app/views/sephora_style_guide/home/_accordion.slim]: ids should be hyphenated: `#collapse-two`

File: app/assets/javascripts/sephora_style_guide/application.js (hunk 1 above)
Comment (KernCheh) [inline app/assets/javascripts/sephora_style_guide/application.js]: oops some historical stuff didn't clear out

File: app/views/sephora_style_guide/home/_accordion.slim (hunk 7 above)
Comment (KernCheh) [inline app/views/sephora_style_guide/home/_accordion.slim]: Hey justin they are essentially slightly different though. Collapsibles are like a component in an accordion. Collapsible can be used out of the accordion context but in this example I'm going straight for an accordion example (which is a collapsible with its header as the collapsible trigger)

File: app/views/sephora_style_guide/home/_accordion.slim (hunk 8 above)
Comment (KernCheh) [inline app/views/sephora_style_guide/home/_accordion.slim]: lol started on it past midnight, blindly copied this from bootstrap

File: app/assets/stylesheets/sephora_style_guide/mixins/_border_radius.scss (hunk 2 above)
Comment (KernCheh) [inline app/assets/stylesheets/sephora_style_guide/mixins/_border_radius.scss]: Unless its ios safari self rendered buttons


jk

**Comment by KernCheh** [comment]
Hey @zamakkat, made some changes already

**Pull request merged**